src/main/java/fworks/da/FtpAccessManager.java

```diff
@@ -131,6 +133,18 @@ public static boolean uploadFile(String localFilePath, String fileName){
         return false;
     }
 
+    @Override
+    public boolean uploadSolutionDocument(SubSDocDsRequestModel model, String docId) {
+        String filePath = model.getFilePath();
+        return uploadFile(filePath, docId);
+    }
+
+    @Override
+    public boolean uploadTestDocument(SubTDocDsRequestModel dsRequestModel, String docId) {
```

Comment:
I think it would make more sense for these two methods to be default methods in the FileAccessGateway. General idea is that the FileAccessGateway should parse the input data (i.e. the request model) in a default method that implements the use case gateway and then call the uploadFile method that is implemented by the FtpAccessManager. This means that you could implement a different concrete file upload system (e.g. S3) without changing the FileAccessGateway but also without the FtpAccessManager having to depend on anything in the use case layer (i.e. the request models).